Replace PseudoIdSet with EnumSet<PseudoId>

anttijk opened the pull request:
#### 0471d4f78defe8c15721b1d8dcdbc52307be35b3
<pre>
Replace PseudoIdSet with EnumSet&lt;PseudoId&gt;
<a href="https://bugs.webkit.org/show_bug.cgi?id=300396">https://bugs.webkit.org/show_bug.cgi?id=300396</a>
<a href="https://rdar.apple.com/162214153">rdar://162214153</a>

Reviewed by Sam Weinig.

Add a generic EnumSet class and use it for PseudoIds.

EnumSet has the same interface as OptionSet but does not require using bitmask style enum.
This makes it more convenient to use especially when compact storage for individual values is also important.

Tests: Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
       Tools/TestWebKitAPI/Tests/WTF/EnumSet.cpp
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/EnumSet.h: Added.
(WTF::requires):
(WTF::EnumSet::storageSize):
(WTF::EnumSet::Iterator::operator* const):
(WTF::EnumSet::Iterator::operator++):
(WTF::EnumSet::Iterator::Iterator):
(WTF::EnumSet::fromRaw):
(WTF::EnumSet::EnumSet):
(WTF::EnumSet::toRaw const):
(WTF::EnumSet::isEmpty const):
(WTF::EnumSet::begin const):
(WTF::EnumSet::end const):
(WTF::EnumSet::operator bool const):
(WTF::EnumSet::contains const):
(WTF::EnumSet::containsAny const):
(WTF::EnumSet::containsAll const):
(WTF::EnumSet::containsOnly const):
(WTF::EnumSet::add):
(WTF::EnumSet::remove):
(WTF::EnumSet::set):
(WTF::EnumSet::hasExactlyOneBitSet const):
(WTF::EnumSet::toSingleValue const):
(WTF::EnumSet::operator|):
(WTF::EnumSet::operator|=):
(WTF::EnumSet::operator&amp;):
(WTF::EnumSet::operator-):
(WTF::EnumSet::operator^):
(WTF::EnumSet::storageMemoryOffset):
(WTF::EnumSet::get const):
* Source/WTF/wtf/Forward.h:
* Source/WebCore/animation/WebAnimationUtilities.h:
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::match const):
(WebCore::SelectorChecker::matchHostPseudoClass const):
(WebCore::hasViewTransitionPseudoElement):
(WebCore::hasScrollbarPseudoElement):
(WebCore::SelectorChecker::matchRecursively const):
(WebCore::SelectorChecker::checkOne const):
(WebCore::SelectorChecker::matchSelectorList const):
(WebCore::SelectorChecker::matchHasPseudoClass const):
* Source/WebCore/css/SelectorChecker.h:
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::SelectorCodeGenerator::generateRequestedPseudoElementEqualsToSelectorPseudoElement):
(WebCore::SelectorCompiler::SelectorCodeGenerator::generateMarkPseudoStyleForPseudoElement):
* Source/WebCore/inspector/agents/InspectorCSSAgent.cpp:
(WebCore::InspectorCSSAgent::getMatchedStylesForNode):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::highlightSelector):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::getCachedPseudoStyle const):
(WebCore::RenderElement::getUncachedPseudoStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::RenderStyle):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleConstants.h:
(WebCore::PseudoIdSet::PseudoIdSet): Deleted.
(WebCore::PseudoIdSet::fromMask): Deleted.
(WebCore::PseudoIdSet::has const): Deleted.
(WebCore::PseudoIdSet::add): Deleted.
(WebCore::PseudoIdSet::remove): Deleted.
(WebCore::PseudoIdSet::merge): Deleted.
(WebCore::PseudoIdSet::operator &amp; const): Deleted.
(WebCore::PseudoIdSet::operator | const): Deleted.
(WebCore::PseudoIdSet::operator bool const): Deleted.
(WebCore::PseudoIdSet::data const): Deleted.
(WebCore::PseudoIdSet::dataMemoryOffset): Deleted.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::pseudoElementType const):
(WebCore::RenderStyle::NonInheritedFlags::hasPseudoStyle const):
(WebCore::RenderStyle::NonInheritedFlags::hasAnyPublicPseudoStyles const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setHasPseudoStyles):
(WebCore::RenderStyle::NonInheritedFlags::setHasPseudoStyles):
(WebCore::RenderStyle::setPseudoElementType):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::ElementRuleCollector::ruleMatches):
* Source/WebCore/style/ElementRuleCollector.h:
(WebCore::Style::ElementRuleCollector::matchedPseudoElementIds const):
* Source/WebCore/style/PseudoElementIdentifier.h:
(WTF::HashTraits&lt;WebCore::Style::PseudoElementIdentifier&gt;::emptyValue):
(WTF::HashTraits&lt;WebCore::Style::PseudoElementIdentifier&gt;::constructDeletedValue):
(WTF::HashTraits&lt;WebCore::Style::PseudoElementIdentifier&gt;::isDeletedValue):
(WTF::HashTraits&lt;std::optional&lt;WebCore::Style::PseudoElementIdentifier&gt;&gt;::emptyValue):
(WTF::HashTraits&lt;std::optional&lt;WebCore::Style::PseudoElementIdentifier&gt;&gt;::constructDeletedValue):
(WTF::HashTraits&lt;std::optional&lt;WebCore::Style::PseudoElementIdentifier&gt;&gt;::isDeletedValue):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WTF/EnumSet.cpp: Added.
(TestWebKitAPI::TEST(WTF_EnumSet, EmptySet)):
(TestWebKitAPI::TEST(WTF_EnumSet, StorageSize)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsOneFlag)):
(TestWebKitAPI::TEST(WTF_EnumSet, Equal)):
(TestWebKitAPI::TEST(WTF_EnumSet, NotEqual)):
(TestWebKitAPI::TEST(WTF_EnumSet, Or)):
(TestWebKitAPI::TEST(WTF_EnumSet, OrAssignment)):
(TestWebKitAPI::TEST(WTF_EnumSet, Minus)):
(TestWebKitAPI::TEST(WTF_EnumSet, AddAndRemove)):
(TestWebKitAPI::TEST(WTF_EnumSet, Set)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsTwoFlags)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsTwoFlags2)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsTwoFlags3)):
(TestWebKitAPI::TEST(WTF_EnumSet, EmptyEnumSetToRawValueToEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsOneFlagToRawValueToEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsOneFlagToRawValueToEnumSet2)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsTwoFlagsToRawValueToEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsTwoFlagsToRawValueToEnumSet2)):
(TestWebKitAPI::TEST(WTF_EnumSet, TwoIteratorsIntoSameEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, IterateOverEnumSetThatContainsTwoFlags)):
(TestWebKitAPI::TEST(WTF_EnumSet, IterateOverEnumSetThatContainsFlags2)):
(TestWebKitAPI::TEST(WTF_EnumSet, NextItemAfterLargestIn32BitFlagSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, NextItemAfterLargestIn64BitFlagSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, IterationOrderTheSameRegardlessOfInsertionOrder)):
(TestWebKitAPI::TEST(WTF_EnumSet, OperatorAnd)):
(TestWebKitAPI::TEST(WTF_EnumSet, OperatorXor)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsAny)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsAll)):
(TestWebKitAPI::TEST(WTF_EnumSet, ToSingleValue)):

Canonical link: <a href="https://commits.webkit.org/301450@main">https://commits.webkit.org/301450@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d704d5ce060f9acfa3549c943f766d97735ef314

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125958 "5 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45618 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36388 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132822 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77812 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/adb8e51d-901c-43e9-9c37-d6b82c3064c0) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127829 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46300 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54174 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95962 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64056 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/36cbb0f2-2350-413f-8237-6b430a1d9e57) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128906 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37033 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112650 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76451 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bce53e0a-21a4-4f17-81e2-574a34f2f0dd) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35941 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30836 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76293 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/118035 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106815 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31049 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135509 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/124465 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52739 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40469 "Found 1 new test failure: fast/webgpu/type-checker-array-without-argument.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104436 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53188 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108863 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104156 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26539 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49538 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27864 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50117 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52633 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58448 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/157478 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51977 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39432 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/js-wasm-js-varying-arities.js.wasm-eager (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55325 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53674 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Tools/TestWebKitAPI/Tests/WTF/EnumSet.cpp:
```diff
@@ -0,0 +1,594 @@
+/*
+ * Copyright (C) 2016-2025 Apple Inc. All rights reserved.
```
Probably don't need this big a range of years.

weinig commented on Source/WTF/wtf/EnumSet.h:
```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2016-2025 Apple Inc. All rights reserved.
```
Probably don't need this big a range of years.

weinig commented on Source/WTF/wtf/EnumSet.h:
```diff
@@ -0,0 +1,213 @@
+// Unlike with OptionSet the enumerators don't need be powers of two but the highest value must be less than 64.
+// If the enum has a member called HighestEnumValue it will be used to compute the storage size. Otherwise 64 bits are used.
+template<typename E> class EnumSet {
+    static_assert(std::is_enum<E>::value, "E is not an enum type");
```
Would recommend using std::is_enum_v<E> here instead.

weinig commented on Source/WTF/wtf/EnumSet.h:
```diff
@@ -0,0 +1,213 @@
+    using StorageType = SizedUnsignedTrait<storageSize()>::Type;
+
+    template<typename StorageType> class Iterator {
+        WTF_DEPRECATED_MAKE_FAST_ALLOCATED(Iterator);
```
Will an iterator ever be allocated? Is this needed?

anttijk replied:
No

weinig commented on Source/WTF/wtf/EnumSet.h:
```diff
@@ -0,0 +1,213 @@
+
+        Iterator& operator++(int) = delete;
+
+        friend bool operator==(const Iterator&, const Iterator&) = default;
```
Trying to get people to use the new shorter version of this:

```
bool operator==(const Iterator&) const = default;
```

weinig commented on Source/WTF/wtf/Forward.h:
```diff
@@ -134,6 +134,7 @@ template<typename T, typename RawValue = uint64_t> using ObjectIdentifier = Obje
 template<typename T, typename RawValue = uint64_t> using AtomicObjectIdentifier = ObjectIdentifierGeneric<T, ObjectIdentifierThreadSafeAccessTraits<RawValue>, RawValue>;
 template<typename> class Observer;
 template<typename, ConcurrencyTag = ConcurrencyTag::None> class OptionSet;
+template<typename> class EnumSet;
```
I think these are supposed to be alphabetical (or mostly anyway).

weinig reviewed (approved):
Really nice!

anttijk committed:
Replace PseudoIdSet with EnumSet<PseudoId>
https://bugs.webkit.org/show_bug.cgi?id=300396
rdar://162214153

Reviewed by Sam Weinig.

Add a generic EnumSet class and use it for PseudoIds.

EnumSet has the same interface as OptionSet but does not require using bitmask style enum.
This makes it more convenient to use especially when compact storage for individual values is also important.

Tests: Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj
       Tools/TestWebKitAPI/Tests/WTF/EnumSet.cpp
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/EnumSet.h: Added.
(WTF::requires):
(WTF::EnumSet::storageSize):
(WTF::EnumSet::Iterator::operator* const):
(WTF::EnumSet::Iterator::operator++):
(WTF::EnumSet::Iterator::Iterator):
(WTF::EnumSet::fromRaw):
(WTF::EnumSet::EnumSet):
(WTF::EnumSet::toRaw const):
(WTF::EnumSet::isEmpty const):
(WTF::EnumSet::begin const):
(WTF::EnumSet::end const):
(WTF::EnumSet::operator bool const):
(WTF::EnumSet::contains const):
(WTF::EnumSet::containsAny const):
(WTF::EnumSet::containsAll const):
(WTF::EnumSet::containsOnly const):
(WTF::EnumSet::add):
(WTF::EnumSet::remove):
(WTF::EnumSet::set):
(WTF::EnumSet::hasExactlyOneBitSet const):
(WTF::EnumSet::toSingleValue const):
(WTF::EnumSet::operator|):
(WTF::EnumSet::operator|=):
(WTF::EnumSet::operator&):
(WTF::EnumSet::operator-):
(WTF::EnumSet::operator^):
(WTF::EnumSet::storageMemoryOffset):
(WTF::EnumSet::get const):
* Source/WTF/wtf/Forward.h:
* Source/WebCore/animation/WebAnimationUtilities.h:
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::match const):
(WebCore::SelectorChecker::matchHostPseudoClass const):
(WebCore::hasViewTransitionPseudoElement):
(WebCore::hasScrollbarPseudoElement):
(WebCore::SelectorChecker::matchRecursively const):
(WebCore::SelectorChecker::checkOne const):
(WebCore::SelectorChecker::matchSelectorList const):
(WebCore::SelectorChecker::matchHasPseudoClass const):
* Source/WebCore/css/SelectorChecker.h:
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::SelectorCodeGenerator::generateRequestedPseudoElementEqualsToSelectorPseudoElement):
(WebCore::SelectorCompiler::SelectorCodeGenerator::generateMarkPseudoStyleForPseudoElement):
* Source/WebCore/inspector/agents/InspectorCSSAgent.cpp:
(WebCore::InspectorCSSAgent::getMatchedStylesForNode):
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
(WebCore::InspectorDOMAgent::highlightSelector):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::getCachedPseudoStyle const):
(WebCore::RenderElement::getUncachedPseudoStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::RenderStyle):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleConstants.h:
(WebCore::PseudoIdSet::PseudoIdSet): Deleted.
(WebCore::PseudoIdSet::fromMask): Deleted.
(WebCore::PseudoIdSet::has const): Deleted.
(WebCore::PseudoIdSet::add): Deleted.
(WebCore::PseudoIdSet::remove): Deleted.
(WebCore::PseudoIdSet::merge): Deleted.
(WebCore::PseudoIdSet::operator & const): Deleted.
(WebCore::PseudoIdSet::operator | const): Deleted.
(WebCore::PseudoIdSet::operator bool const): Deleted.
(WebCore::PseudoIdSet::data const): Deleted.
(WebCore::PseudoIdSet::dataMemoryOffset): Deleted.
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::pseudoElementType const):
(WebCore::RenderStyle::NonInheritedFlags::hasPseudoStyle const):
(WebCore::RenderStyle::NonInheritedFlags::hasAnyPublicPseudoStyles const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setHasPseudoStyles):
(WebCore::RenderStyle::NonInheritedFlags::setHasPseudoStyles):
(WebCore::RenderStyle::setPseudoElementType):
* Source/WebCore/style/ElementRuleCollector.cpp:
(WebCore::Style::ElementRuleCollector::ruleMatches):
* Source/WebCore/style/ElementRuleCollector.h:
(WebCore::Style::ElementRuleCollector::matchedPseudoElementIds const):
* Source/WebCore/style/PseudoElementIdentifier.h:
(WTF::HashTraits<WebCore::Style::PseudoElementIdentifier>::emptyValue):
(WTF::HashTraits<WebCore::Style::PseudoElementIdentifier>::constructDeletedValue):
(WTF::HashTraits<WebCore::Style::PseudoElementIdentifier>::isDeletedValue):
(WTF::HashTraits<std::optional<WebCore::Style::PseudoElementIdentifier>>::emptyValue):
(WTF::HashTraits<std::optional<WebCore::Style::PseudoElementIdentifier>>::constructDeletedValue):
(WTF::HashTraits<std::optional<WebCore::Style::PseudoElementIdentifier>>::isDeletedValue):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WTF/EnumSet.cpp: Added.
(TestWebKitAPI::TEST(WTF_EnumSet, EmptySet)):
(TestWebKitAPI::TEST(WTF_EnumSet, StorageSize)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsOneFlag)):
(TestWebKitAPI::TEST(WTF_EnumSet, Equal)):
(TestWebKitAPI::TEST(WTF_EnumSet, NotEqual)):
(TestWebKitAPI::TEST(WTF_EnumSet, Or)):
(TestWebKitAPI::TEST(WTF_EnumSet, OrAssignment)):
(TestWebKitAPI::TEST(WTF_EnumSet, Minus)):
(TestWebKitAPI::TEST(WTF_EnumSet, AddAndRemove)):
(TestWebKitAPI::TEST(WTF_EnumSet, Set)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsTwoFlags)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsTwoFlags2)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsTwoFlags3)):
(TestWebKitAPI::TEST(WTF_EnumSet, EmptyEnumSetToRawValueToEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsOneFlagToRawValueToEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsOneFlagToRawValueToEnumSet2)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsTwoFlagsToRawValueToEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, EnumSetThatContainsTwoFlagsToRawValueToEnumSet2)):
(TestWebKitAPI::TEST(WTF_EnumSet, TwoIteratorsIntoSameEnumSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, IterateOverEnumSetThatContainsTwoFlags)):
(TestWebKitAPI::TEST(WTF_EnumSet, IterateOverEnumSetThatContainsFlags2)):
(TestWebKitAPI::TEST(WTF_EnumSet, NextItemAfterLargestIn32BitFlagSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, NextItemAfterLargestIn64BitFlagSet)):
(TestWebKitAPI::TEST(WTF_EnumSet, IterationOrderTheSameRegardlessOfInsertionOrder)):
(TestWebKitAPI::TEST(WTF_EnumSet, OperatorAnd)):
(TestWebKitAPI::TEST(WTF_EnumSet, OperatorXor)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsAny)):
(TestWebKitAPI::TEST(WTF_EnumSet, ContainsAll)):
(TestWebKitAPI::TEST(WTF_EnumSet, ToSingleValue)):

Canonical link: https://commits.webkit.org/301450@main
